src/rate.rs

```diff
@@ -394,6 +394,9 @@ pub struct RCState {
   // The maximum buffer fullness (total size of the buffer).
   reservoir_max: i64,
   // The log of estimated scale factor for the rate model in Q57 format.
+  //
+  // TODO: Convert to Q23 or figured out a better way to avoid overflow
+  // once 2-pass mode is introduced.
```

Comment:
Typo with "figured"  rather than "figure". I recommend qualifying with an "if required" or similar  somewhere in this comment.

Reply:
Done.